android/app/src/main/java/updated/mysterium/vpn/database/entity/NodeEntity.kt

```diff
@@ -0,0 +1,41 @@
+package updated.mysterium.vpn.database.entity
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import network.mysterium.service.core.ProposalItem
+
+@Entity
+data class NodeEntity(
+    @PrimaryKey val id: String,
+    @ColumnInfo(name = "provider_id") val providerID: String,
+    @ColumnInfo(name = "service_type") val serviceType: String,
+    @ColumnInfo(name = "country_code") val countryCode: String,
+    @ColumnInfo(name = "node_type") val nodeType: String,
+    @ColumnInfo(name = "monitoring_failed") val monitoringFailed: Boolean,
+    @ColumnInfo(name = "payment_type") val paymentType: String,
+    @ColumnInfo(name = "payment_amount") val paymentAmount: Double,
+    @ColumnInfo(name = "currency") val currency: String,
+    @ColumnInfo(name = "price_per_second") val pricePerSecond: Double,
+    @ColumnInfo(name = "price_per_byte") val pricePerByte: Double,
+    @ColumnInfo(name = "quality_level") var qualityLevel: Int,
+) {
+
+    companion object {
+
+        fun createNodeFromProposal(proposalItem: ProposalItem) = NodeEntity(
```

Comment:
Instead of creating static methods create custom constructor. See an example in other comment.

Reply:
Fix